security/self-assessment.md

```diff
@@ -0,0 +1,488 @@
+  case of Python SDK and `kubectl`, Katib relies on Kubernetes RBAC, in case of Kubeflow Dashboard
+  user access it via Istio gateway.
+
+- Security and Access Control: Katib leverages Kubernetes RBAC to ensure that user can
+  manage Experiments.
+
+### Kubeflow Trainer
+
+- TrainJob creation: User creates TrainJob CRDs, specifying configuration for the training job.
+  User should create TrainJob using Python SDK. Python SDK relies on Kubernetes RBAC to access
+  Kubernetes cluster.
+
+- TrainingRuntime creation: User creates TrainingRuntime or ClusterTrainingRuntime CRDs, specifying
+  template configuration for the training job. User should create CRDs via `kubectl`.
+
+- Security and Access Control: Kubeflow Trainer leverages Kubernetes RBAC to ensure that user can
+  manage TrainJobs and TrainingRuntimes.
+
+### Kubeflow Model Registry
+
+- Save models: User utilizes Python client to interact with Model Registry. Also, user can use
+  Kubeflow Dashboard to manage models. In that case, user access it via Istio gateway
+
+### Kubeflow Pipelines
+
+- Create pipelines: User utilized Python client to manage ML pipelines. Also,
+  can use Kubeflow Dashboard to access KFP UI. In that case, user access it via Istio gateway
+
+- Security and Access Control: Kubeflow Pipelines leverages Kubernetes RBAC to ensure that user can
+  manage Pipelines.
+
+## Goals
+
+- **Foundation for AI Platforms on Kubernetes**: Kubeflow projects provide a standard, cloud native
+  solution to build end-to-end AI platforms on Kubernetes. The Kubeflow tools are scalable and
+  composable which give organizations flexibility to adopt them.
+
+- **Simplify Complexity**: Abstract the Kubernetes complexity for ML users to manage and scale AI
+  workloads. Define Python-first experience for users, so they can focus on their models rather
+  than the underlying infrastructure.
+
+- **Large Scale Model Development**: Provide capability to develop, train, optimize, and deploy
+  LLMs at scale.
+
+- **Out of the Box ML Framework Support**: Support a wide range of ML frameworks like PyTorch,
+  JAX, HuggingFace, DeepSpeed, XGboost, and others to run AI workloads.
+
+- **Advanced Orchestration Patterns**: Support advanced orchestration patterns such as gang-scheduling,
+  High-Performance Computing jobs, GPU cost optimization to run AI workloads.
+
+- **End-to-End GenAI on Kubernetes**: Provides simple APIs and interfaces for users to perform
+  end-to-end GenAI use-cases including LLM fine-tuning, RAG, LLM optimizations, LLM inference,
+  AI agent orchestration.
+
+- **Interactive AI development**: Provide interactive development environment for ML users to
+  develop and deploy AI models.
+
+## Non-Goals
+
+- Target non-Kubernetes deployment systems.
+- Kubeflow doesn't re-invent ML frameworks and packages (e.g. PyTorch, JAX).
+- Kubeflow is not replacement for GitOps systems like ArgoCD.
+- Kubeflow doesn't enforce a deployment method or distribution for Kubeflow projects.
```

Comment:
I think we do, Kustomize and helm

Reply:
I think, @thesuperzapper added this no-goal to the GTR doc: https://docs.google.com/document/d/15CZtkk3x-YIUaNnaRzIZaIKrfDfT6is_PAlRDIoBKgQ/edit?tab=t.0

@thesuperzapper Can you give context here please ?